### PR TITLE
Doc guideline compliance

### PR DESCRIPTION
**Pull request by outoftardis**
# Description
- Removed known issues (based on the email from Nikolay)
- Changed the order of win and lin sections in Build app w/ onedal + changed the wording of the first step about installation (also based on the request from Nikolay)
- Made changes to comply with the requirements for oneapi open source docs. This is mainly about doc structure, removing links to idz (some are still left though), adding a sys requirements page (yes, it copies the content from idz but we are not supposed to refer there)
- Made changes that assure that we can create a version of this doc for IDZ (no contirbuting guide, links to idz will be there, installation from base toolkit -- all that will be preserved in that version of onedal docs)

output: https://outoftardis.github.io/daal

### CODE REVIEW

**Comment by outoftardis** [comment]
@agorshk @makart19 please review

File: docs/source/conf.py [hunk 1]
```diff
@@ -43,21 +43,15 @@
...
+.. include:: /substitutions_common.txt
+.. include:: /substitutions_specific.txt
 """
 
-if tags.has('use_intelname'):
-    # use supplied =t use_intelname
-    rst_prolog += ".. |full_name| replace:: Intel\ |reg|\  oneAPI Data Analytics Library"
-else:
-    rst_prolog += ".. |full_name| replace:: oneAPI Data Analytics Library"
+# if tags.has('use_intelname'):
```
Comment (makart19) [inline docs/source/conf.py:50]: comments need to be removed?

**Review by makart19 — APPROVED** [review]

File: docs/source/system-requirements.rst [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+      - Intel® Iris® Graphics
+      - Intel® Iris® Pro Graphics
+
+   .. tab:: CPU
+
+      - Intel Atom® Processors
+      - Intel® Core™ Processor Family
+      - Intel® Xeon® Processor Family
+      - Intel® Xeon® Scalable Performance Processor Family
+
```
Comment (napetrov) [inline docs/source/system-requirements.rst:40]: Let's add "Generic X86 processor"  - or might be other proper way to name non IA CPUs

File: docs/source/system-requirements.rst [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+Software Requirements
+*********************
+
+- GCC* 7.x or higher or Intel® C++ Compiler 19.1 and later
+- Intel® oneAPI DPC++ Compiler latest release (for oneAPI DPC++ interfaces)
+- Intel® oneAPI Threading Building Blocks latest release (for the multi-threaded version of oneDAL)
+- OpenCL(TM) 1.2 or later
```
Comment (napetrov) [inline docs/source/system-requirements.rst:47]: Let's drop OpenCL

File: docs/source/system-requirements.rst [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+- C/C++ Compiler with C++11 support (or C++14 support on Windows*)
+- Microsoft Visual Studio* (2019 and 2022 versions) needed only if using Visual Studio IDE for development
+
+Operating Systems:
+
+- 64-bit Linux*: Ubuntu* 18.04 or higher
+- 64-bit Windows* 10
```
Comment (napetrov) [inline docs/source/system-requirements.rst:54]: Suggest to just mention that library support 64-bit only and do not mention this for each build.

For windows - let's add Windows Server 2019  or higher.  
For Win 10 - + or higher as we support Win11 

On linux - Let's point that in theory we will support any linux with certain GLIBC/BINUTILS requirements , and mention that we are testing RHEL and UBUNTU.
@maria-Petrova @homksei  @vmeshche  can comment

File: docs/source/onedal/build_app/build-application.rst [hunk 5]
```diff
@@ -17,21 +17,81 @@
...
+Applications on Linux* OS
+-------------------------
+
+#. Install |short_name|.
+
+#. Set environment variables by calling ``<install dir>/setvars.sh``.
+
+#. Build your application with clang++:
+
+   - Add ``fsycl`` option to the command:
```
Comment (napetrov) [inline docs/source/onedal/build_app/build-application.rst:36]: @makart19 @amgrigoriev @vlad-nazarov @KulikovNikita  - what would  be correct instructions?

I think we have to replace clang++ with dpcpp call without fsycl. Also don;t sure that "ONEAPI_DAAL_USE_MKL_GPU_GEMM" is required

File: docs/source/onedal/build_app/build-application.rst [hunk 6]
```diff
@@ -17,21 +17,81 @@
 Build applications with oneDAL
 ==============================
 
-See Get Started Guides for `Windows\*
-<https://software.intel.com/content/www/us/en/develop/documentation/get-started-with-daal-for-windows/top.html>`_
-and `Linux\* <https://software.intel.com/content/www/us/en/develop/documentation/get-started-with-daal-for-linux/top.html>`_
-for instruction on how to build applications for C++.
 This section contains instructions for building applications with |short_name| for SYCL\*.
 
-- :ref:`app_on_win`
```
Comment (agorshk) [inline docs/source/onedal/build_app/build-application.rst:26]: what about mac?

File: docs/source/system-requirements.rst (hunk 3 above)
Comment (outoftardis) [reply docs/source/system-requirements.rst]: done

File: docs/source/system-requirements.rst (hunk 2 above)
Comment (outoftardis) [reply docs/source/system-requirements.rst:41]: done

File: docs/source/conf.py (hunk 1 above)
Comment (outoftardis) [reply docs/source/conf.py]: done

File: docs/source/onedal/build_app/build-application.rst (hunk 6 above)
Comment (outoftardis) [reply docs/source/onedal/build_app/build-application.rst:26]: well I only restructured what we had, and we didn't have anything for mac.
I'd suggest adding instructions for mac in a separate PR

File: docs/source/system-requirements.rst (hunk 4 above)
Comment (outoftardis) [reply docs/source/system-requirements.rst]: done except for this:

> On linux - Let's point that in theory we will support any linux with certain GLIBC/BINUTILS requirements , and mention that we are testing RHEL and UBUNTU.
@maria-Petrova @homksei @vmeshche can comment

not sure what I need to add

File: docs/source/onedal/build_app/build-application.rst (hunk 5 above)
Comment (outoftardis) [reply docs/source/onedal/build_app/build-application.rst:36]: @makart19 @amgrigoriev @vlad-nazarov @KulikovNikita can anyone comment on this?

**Comment by outoftardis** [comment]
@napetrov @agorshk  are you ok with merging these changes? I understand that there are open questions left but they are not really connected to the main purpose of the PR that was to align docs and comply with the doc guidelines for oneAPI libs

File: docs/source/onedal/build_app/build-application.rst (hunk 5 above)
Comment (napetrov) [reply docs/source/onedal/build_app/build-application.rst:36]: @makart19 @amgrigoriev @vlad-nazarov @KulikovNikita can anyone comment on this?

File: docs/source/system-requirements.rst (hunk 4 above)
Comment (napetrov) [reply docs/source/system-requirements.rst]: thanks

File: docs/source/onedal/build_app/build-application.rst (hunk 5 above)
Comment (vlad-nazarov) [reply docs/source/onedal/build_app/build-application.rst:36]: I think it is better to replace clang++ with dpcpp and remove -fsycl flag. Our examples we build in this way.

Comment (vlad-nazarov) [reply docs/source/onedal/build_app/build-application.rst:36]: I didn't find any place in our code where "ONEAPI_DAAL_USE_MKL_GPU_GEMM" export would be used.

Comment (vlad-nazarov) [reply docs/source/onedal/build_app/build-application.rst:36]: Let me check the relevance of the entire instruction within a few days. We have added the libonedal_dpc library, which is not listed here

Comment (outoftardis) [reply docs/source/onedal/build_app/build-application.rst:36]: @vlad-nazarov  please let me know when you check the instructions

Comment (vlad-nazarov) [reply docs/source/onedal/build_app/build-application.rst]: @outoftardis, updated the Linux example. Please, check. Maybe should we do the same for windows? I think windows instruction not actual too...

Comment (outoftardis) [reply docs/source/onedal/build_app/build-application.rst]: @vlad-nazarov @makart19 @KulikovNikita @amgrigoriev @napetrov who could tell whether any changes are needed for Windows instructions?

**Review by napetrov — APPROVED** [review]

**Pull request merged**